### ISSUE TITLE
NPM Registry does not work with proxy

### ISSUE THREAD

**Issue by swright787** [DependencyTrack/dependency-track#258]
### Issue Type:
  - [x] defect report
  - [ ] enhancement request
  
  
### Current Behavior:
When using OWASP Dependency Track behind a proxy, the NIST vulnerability database is downloaded, however the connection to the NPM Registy does not work. The connection times out, which makes it likely that it is not using the configured proxy.

### Steps to Reproduce (if defect):
Configure OWASP Dependency Track to use a proxy (and not have a direct connection out to https://registry.npmjs.org). After starting OWASP Depedency Track observe the following Exception in the .dependency-track/dependency-track.log logs:

```
io.github.openunirest.http.exceptions.UnirestException: org.apache.http.conn.ConnectTimeoutException: Connect to registry.npmjs.org:443 [registry.npmjs.org/104.16.23.35, registry.npmjs.org/104.16.21.35, registry.npmjs.org/104.16.20.35, registry.npmjs.org/104.16.17.35, registry.npmjs.org/104.16.26.35, registry.npmjs.org/104.16.22.35, registry.npmjs.org/104.16.25.35, registry.npmjs.org/104.16.27.35, registry.npmjs.org/104.16.24.35, registry.npmjs.org/104.16.16.35, registry.npmjs.org/104.16.19.35, registry.npmjs.org/104.16.18.35] failed: connect timed out
        at io.github.openunirest.request.HttpClientHelper.request(HttpClientHelper.java:38)
        at io.github.openunirest.request.BaseRequest.asJson(BaseRequest.java:71)
        at org.dependencytrack.tasks.NpmAdvisoryMirrorTask.getAdvisories(NpmAdvisoryMirrorTask.java:98)
        at org.dependencytrack.tasks.NpmAdvisoryMirrorTask.inform(NpmAdvisoryMirrorTask.java:67)
        at alpine.event.framework.BaseEventService.lambda$publish$0(BaseEventService.java:96)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.http.conn.ConnectTimeoutException: Connect to registry.npmjs.org:443 [registry.npmjs.org/104.16.23.35, registry.npmjs.org/104.16.21.35, registry.npmjs.org/104.16.20.35, registry.npmjs.org/104.16.17.35, registry.npmjs.org/104.16.26.35, registry.npmjs.org/104.16.22.35, registry.npmjs.org/104.16.25.35, registry.npmjs.org/104.16.27.35, registry.npmjs.org/104.16.24.35, registry.npmjs.org/104.16.16.35, registry.npmjs.org/104.16.19.35, registry.npmjs.org/104.16.18.35] failed: connect timed out
        at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:151)
        at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:373)
        at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:394)
        at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:237)
        at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)
        at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
        at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
        at io.github.openunirest.request.HttpClientHelper.request(HttpClientHelper.java:33)
        ... 7 common frames omitted
Caused by: java.net.SocketTimeoutException: connect timed out
        at java.net.PlainSocketImpl.socketConnect(Native Method)
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.net.Socket.connect(Socket.java:589)
        at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:339)
        at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
        ... 18 common frames omitted
```

### Expected Behavior:

The connection to the NPM registry should use the configured proxy to be able to download the mirror the NPM database via the configured proxy.

### Environment:
  
  - Dependency-Track Version: v3.3.1
  - Distribution: Traditional WAR 
  - BOM Format & Version: N/A
  - Database Server: MySQL
  - Browser: N/A

**Comment by swright787** [comment]
I have confirmed that using curl with the same proxy on the same server, I was able to retrieve the JSON from https://registry.npmjs.org/-/npm/v1/security/advisories

**Comment by swright787** [comment]
I think it may actually be a bug with Unirest: https://github.com/OpenUnirest/unirest-java/issues/42

**Comment by stevespringett** [comment]
As part of the release process, components are updated. OpenUnirest is being updated to 2.5.03 so this issue should be resolved in the upcoming 3.4.0 release.
